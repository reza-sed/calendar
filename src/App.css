* {
  box-sizing: border-box;
}

body {
  font-family: Roboto, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
    Oxygen, Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  background: var(--color-secondary);
}

/* App global color pallete */
:root {
  --color-primary: #aeaeae;
  --color-secondary: #e4e4e4;
  --color-accent: #d4665b;
  /* events global color set */
  --color-event-success: #5cb058;
  --color-event-danger: #b02942;
  --color-event-warn: #d35c27;
}

/* define varous events related type colors */
.event--success {
  fill: var(--color-event-success);
}

.event--danger {
  fill: var(--color-event-danger);
}

.event--warn {
  fill: var(--color-event-warn);
}
