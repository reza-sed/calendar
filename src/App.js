import "./App.css";
import Calendar from "./component/calendar";

// events status
export const eventStatus = {
  success: "event--success",
  danger: "event--danger",
  warn: "event--warn",
};
// events structure
const events = [
  { id: 1, desc: "event 1", status: eventStatus.success, date: "2021-07-01" },
  { id: 2, desc: "event 2", status: eventStatus.danger, date: "2021-07-01" },
  { id: 3, desc: "event 3", status: eventStatus.success, date: "2021-07-01" },
  { id: 4, desc: "event 4", status: eventStatus.warn, date: "2021-07-02" },
];

function App() {
  return (
    <div className="App">
      <Calendar month={6} year={2021} events={events} />
    </div>
  );
}

export default App;
