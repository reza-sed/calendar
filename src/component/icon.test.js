import React from "react";
import renderer from "react-test-renderer";
import ICON_NAMES from "../logic/contants";
import Icon from "./icon";
import { eventStatus } from "../App";

// events status

describe("Icon component", () => {
  it("snapshot testing with custom cls for colorizing", () => {
    const component = renderer.create(
      <Icon name={ICON_NAMES.EVENT} cls={eventStatus.success} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("snapshot testing without custom cls", () => {
    const component = renderer.create(<Icon name={ICON_NAMES.EVENT} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
