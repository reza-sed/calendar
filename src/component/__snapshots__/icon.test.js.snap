// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component snapshot testing with custom cls for colorizing 1`] = `
<span
  className="icon"
  id="circle"
>
  <svg
    className="icon__element event--success"
  >
    <use
      xlinkHref="/images/icons.svg#circle"
    />
  </svg>
</span>
`;

exports[`Icon component snapshot testing without custom cls 1`] = `
<span
  className="icon"
  id="circle"
>
  <svg
    className="icon__element"
  >
    <use
      xlinkHref="/images/icons.svg#circle"
    />
  </svg>
</span>
`;
